src/Twig/RolesMatrixExtension.php

```diff
@@ -0,0 +1,96 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\UserBundle\Twig;
+
+use Sonata\UserBundle\Security\RolesBuilderInterface;
+use Symfony\Component\Form\FormView;
+use Twig_Environment;
+use Twig_Extension;
+use Twig_SimpleFunction;
+
+/**
+ * @author Christian Gripp <mail@core23.de>
+ * @author Cengizhan Çalışkan <cengizhancaliskan@gmail.com>
+ * @author Silas Joisten <silasjoisten@hotmail.de>
+ */
+final class RolesMatrixExtension extends Twig_Extension
+{
+    /**
+     * @var RolesBuilderInterface
+     */
+    private $rolesBuilder;
+
+    public function __construct(RolesBuilderInterface $rolesBuilder)
+    {
+        $this->rolesBuilder = $rolesBuilder;
+    }
+
+    public function getFunctions(): array
+    {
+        return [
+            new Twig_SimpleFunction('renderMatrix', [$this, 'renderMatrix'], ['needs_environment' => true]),
```

Comment:
This is still the old way, you should use:

https://github.com/sonata-project/SonataAdminBundle/blob/3a5f85ee9909c0fa20623394450e06b96174a32c/src/Twig/Extension/SonataAdminExtension.php#L25-L31